pandas/tseries/index.py

```diff
@@ -1202,9 +1202,10 @@ def intersection(self, other):
               not other.offset.isAnchored() or
               (not self.is_monotonic or not other.is_monotonic)):
             result = Index.intersection(self, other)
-            if isinstance(result, DatetimeIndex):
-                if result.freq is None:
-                    result.offset = to_offset(result.inferred_freq)
+            result = self._simple_new(result._values, name=result.name,
```

Comment:
If I use _shallow_copy, the result will inherit all self attributes (e.g., freq) and this might not be the required result. For example, this test would fail:
```
base = date_range('6/1/2000', '6/30/2000', freq='D', name='idx')

rng4 = date_range('7/1/2000', '7/31/2000', freq='D', name='idx') 
expected4 = DatetimeIndex([], name='idx')
```
because result would inherit self freq = 'D'.